### PR TITLE
LPS-127135 fix post organization

### PR DESCRIPTION
**Pull request by SamZiemer**
**Ticket:**
<https://issues.liferay.com/browse/LPS-127135>

**Notes from @kevhlee:**
>Users are not able to create a new organization via the Headless Admin API because `_getDTOConverterContext()` expects an organization ID, not the organization as an object string. Doing this causes an exception, resulting in a 404 error. This fix was originally developed by @javierdearcos from the Liferay Headless Team. I am simply pushing his changes to `master`.
>
>cc: @javierdearcos

### CODE REVIEW

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/OrganizationResourceImpl.java [hunk 1]
```diff
@@ -173,8 +173,8 @@ public Organization postOrganization(Organization organization)
 				ServiceContextFactory.getInstance(contextHttpServletRequest));
 
 		return _organizationResourceDTOConverter.toDTO(
-			_getDTOConverterContext(String.valueOf(serviceBuilderOrganization)),
-			serviceBuilderOrganization);
+			_getDTOConverterContext(
+				serviceBuilderOrganization.getExternalReferenceCode()));
```
Comment (drewbrokke) [inline modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/OrganizationResourceImpl.java:177]: The result of `getExternalReferenceCode` will always be `null` here. We should use `String.valueOf(serviceBuilderOrganization.getOrganizationId())` instead.

Comment (kevhlee) [reply modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/OrganizationResourceImpl.java:177]: I see. Thanks for catching that, @drewbrokke! Would you like us to submit a new pull with the revision?

Comment (drewbrokke) [reply modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/OrganizationResourceImpl.java:177]: Yes please :)

**Comment by drewbrokke** [comment]
@kevhlee can you also include a test case to reproduce this issue? That way we can avoid further regressions going forward.

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/OrganizationResourceImpl.java (hunk 1 above)
Comment (kevhlee) [reply modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/OrganizationResourceImpl.java:177]: Sounds good! We will do that.

**Pull request closed without merging**